datadog-agent: fix extraIntegrations

cransom opened the pull request:
The override that builds the custom python for integrations-core was overriding python, but pythonPackages was still being inherited from a call to `datadog-integrations-core {}`, causing service.datadog-agent.extraIntegrations to be ignored.

###### Motivation for this change

`services.datadog-agent.extraIntegrations` failed to include varnish related files from datadogs `integration-core` repo when I tried to include varnish.


###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

There isn't an integrated nixos test for datadog, but you can run the daemon and validate scripts without a functional API key or network access (assuming the checks don't need external network access). I used this snippet to validate my varnish change succeeded.

```
import <nixpkgs/nixos/tests/make-test.nix> ({ pkgs, ...} : {
  name = "datadog";

  nodes = {
    machine =
      { ... }:
      {
        imports = [ ../modules/profiles/minimal.nix ];
        services.datadog-agent.enable = true;
        users.groups.varnish.members = [ "datadog" ];
        services.datadog-agent.apiKeyFile = "/dev/null";
        services.datadog-agent.extraIntegrations = {
          varnish = (ps: [ ]);
        };
        services.datadog-agent.checks = {
          varnish = {
            init_config = {};
            instances = [{
              varnishstat = "${pkgs.varnish}/bin/varnishstat";
            }];
          };
        };
        services.varnish.enable = true;
        services.varnish.config = ''
          vcl 4.0;
          backend foo {
            .host = "127.0.0.1";
            .port = "8080";
          }
        '';
      };
    };

    testScript = ''
      startAll;
      $machine->waitForUnit("varnish.service");
      $machine->succeed("sleep 30");
      $machine->fail("journalctl -b -u datadog-agent | grep -q 'Error running check varnish'");
    '';
  })
```

Commit:
datadog-agent: fix extraIntegrations

The override that builds the custom python for integrations-core was
overriding python, but pythonPackages was still being inherited from a
call to `datadog-integrations-core {}`, causing
service.datadog-agent.extraIntegrations to be ignored.

JohnAZoidberg commented:
You'll also have to change this doc string: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/monitoring/datadog-agent.nix#L63
And remove those lines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/networking/dd-agent/datadog-agent.nix#L29-L30

cransom commented:
@JohnAZoidberg I switched the doc string (python -> pythonPackages) and removed the comment about needing to override python. The python interpreter will come from the pythonPackages supplied rather than implying it should be something else.

JohnAZoidberg commented:
Ah yeah right, you can't remove the second line. But I think it doesn't need to be inside the derivation - you can move it up into the `let` expression.

cransom commented:
shuffled python into the `let`.

cransom commented:
anything else to address?
